File: circle.py
import math
from figure import Figure

class Circle(Figure):
    def __init__(self, color: str, radius: float):
        self.color = color
        self.radius = radius

    def getColor(self):
        return self.color

    def get_name(self):
        return "Circle"

    def perimeter(self):
        return 2 * math.pi * self.radius

    def area(self):
        return math.pi * (self.radius ** 2)

File: Figure/src/figure/circle.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package figure;

/**
 *
 * @author eyro1
 */
public class circle implements figure {
    private String color;
    private double radius;

    public circle(String color, double radius) {
        this.color = color;
        this.radius = radius;
    }

    @Override
    public String getColor() {
        return color;
    }

    @Override
    public String getName() {
        return "Circle";
    }

    @Override
    public double perimeter() {
        return 2 * Math.PI * radius;
    }

    @Override
    public double area() {
        return Math.PI * Math.pow(radius, 2);
    }
}

